crates/xxi-node/src/node/wallet.rs

```diff
@@ -71,6 +71,9 @@ impl<D: BdkStorage, S: TenTenOneStorage, N: Storage + Send + Sync + 'static> Nod
     /// Sync the state of the on-chain wallet against the blockchain.
     pub async fn sync_on_chain_wallet(&self) -> Result<()> {
         let client = &self.blockchain.esplora_client_async;
+        // On sync, we unlock our locked utxos, because we are syncing with the blockchain we will
+        // find already spent utxos and release those which were locked unnecessarily.
+        self.wallet.locked_utxos.lock().clear();
```

Comment:
🔧 I think it would make more sense to move this to end of this function. The sync is fast, but it's safer to unlock these once the wallet state has been updated.

Reply:
fixed it.